src/Monitor/Monitor/ScheduledQueryRules/SetScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,229 @@
+
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets source (Query, DataSourceId, etc.) for rule.
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule source")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule source")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule source")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSource Source { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets schedule (Frequnecy, Time Window) for rule.
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule schedule")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule schedule")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule schedule")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleSchedule Schedule { get; set; }
+
+        //
+        // Summary:
+        //     Gets or sets action needs to be taken on rule execution.
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule Alerting Action")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule Alerting Action")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The scheduled query rule Alerting Action")]
+        [ValidateNotNullOrEmpty]
+        public PSScheduledQueryRuleAlertingAction Action { get; set; }
+
+        //
+        // Summary:
+        //     Region where alert is to be created
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The location for this alert")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The location for this alert")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The location for this alert")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
```

Comment:
@shpimpal same comment about adding the `LocationCompleter`

Reply:
sure.